Scalameta confused by weirdly-formatted for comprehension

Hi,

When I try to format the following code with scalafmt 3.10.3
```scala
  val live =
    for
    given Foo <- frob
    yield
      identity:
        sql""
```
I get an error:
```
org.scalafmt.dynamic.exceptions.PositionExceptionImpl: /tmp/test.scala:6: error: [dialect scala3 [with overrides]] `identifier` expected but `interpolation id` found
        sql""
        ^: /tmp/test.scala
```

Scala 3.7.4 can parse this. Indenting the `given` line makes the error go away.

--- a/scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScannerTokens.scala
+++ b/scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScannerTokens.scala
@@ -490,7 +490,11 @@ final class ScannerTokens(val tokens: Tokens)(implicit dialect: Dialect) {
         } else currRef(sepRegions)
       case _: KwEnum => currRef(RegionTemplateMark :: sepRegions)
       case _: KwGiven if !prevToken.isAny[Dot, KwCase] =>
-        currRef(new RegionGivenDecl(curr) :: sepRegions)
+        currRef(dropRegionLine(sepRegions) match {
+          case (_: RegionFor) :: _ => sepRegions
+          case (_: RegionDelim) :: (_: RegionFor) :: _ => sepRegions
+          case rs => new RegionGivenDecl(curr) :: rs
+        })
       case _: KwWith => currRef(dropRegionLine(sepRegions) match {
           case (_: RegionGivenDecl) :: rs => rs
           case _ => sepRegions

--- a/tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ControlSyntaxSuite.scala
+++ b/tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ControlSyntaxSuite.scala
@@ -4264,11 +4264,16 @@ class ControlSyntaxSuite extends BaseDottySuite {
          |  identity:
          |    sql""
          |""".stripMargin
-    val error =
-      """|<input>:5: error: `identifier` expected but `interpolation id` found
-         |    sql""
-         |    ^""".stripMargin
-    runTestError[Stat](code, error)
+    val layout =
+      """|for (given Foo <- frob) yield identity {
+         |  sql""
+         |}
+         |""".stripMargin
+    val tree = Term.ForYield(
+      Term.EnumeratorsBlock(List(Enumerator.Generator(Pat.Given("Foo"), "frob"))),
+      tapply("identity", blk(Term.Interpolate("sql", List(lit("")), Nil)))
+    )
+    runTestAssert[Stat](code, layout)(tree)
   }
 
   test("#4421 indented") {

